static/app/views/monitors/monitors.tsx

```diff
@@ -65,37 +53,29 @@ function NewMonitorButton(props: ButtonProps) {
   );
 }
 
-class Monitors extends AsyncView<Props, State> {
-  get orgSlug() {
-    return this.props.organization.slug;
-  }
-
-  getEndpoints(): ReturnType<AsyncView['getEndpoints']> {
-    const {location} = this.props;
-    return [
-      [
-        'monitorList',
-        `/organizations/${this.orgSlug}/monitors/`,
-        {
-          query: {...location.query, includeNew: true},
-        },
-      ],
-    ];
-  }
-
-  getTitle() {
-    return `Crons - ${this.orgSlug}`;
-  }
-
-  onRequestSuccess(response): void {
-    this.props.setEventNames('monitors.page_viewed', 'Monitors: Page Viewed');
-    this.props.setRouteAnalyticsParams({
-      empty_state: response.data.length === 0,
-    });
-  }
-
-  handleSearch = (query: string) => {
-    const {location, router} = this.props;
+export default function Monitors({location}: RouteComponentProps<{}, {}>) {
+  const organization = useOrganization();
+  const router = useRouter();
+  const queryClient = useQueryClient();
+
+  const monitorListQueryKey = [
+    `/organizations/${organization.slug}/monitors/`,
+    {query: {...location.query, includeNew: true}},
+  ] as const;
+  const {
+    data: monitorList,
+    getResponseHeader: monitorListHeaders,
+    isLoading,
+  } = useApiQuery<Monitor[]>(monitorListQueryKey, {
+    staleTime: 0,
+  });
+
+  useRouteAnalyticsEventNames('monitors.page_viewed', 'Monitors: Page Viewed');
```

Comment:
checked with steve, moving these hooks out to the top of the component, instead of onRequestSuccess should be fine. New data will just override old params